README
A Zikula filter hook to display emoticon images

##Version 3.0.0

_Under current development_

Requires Zikula Core 1.3+

docs/changelog.txt
Version 3.0.0
    - refactored for Zikula 1.3.0ff, will not run with previous versions!
    - This version needs a complete re-installation, upgrade from previous versions is not supported

Version 2.1 
===============================================================================
IMPORTANT NOTICE:
==================================================================================
The permission component changed in v2.0 from pn_bbsmile to BBSmile, please update
your permission rules!!!
You might also need to change the path settings that point to your smilies images.
===============================================================================
Upgrade from pn_bbsmile:
* remove old pn_bbsmile module in the filesystem, do not remove the module in
  Zikula!
* copy new BBSmile files to your modules folder
* regenerate modules list and continue upgrading as usual
==================================================================================
    - renamed to BBSmile to get rid of the pn_ prefix.

Version 2.0 (not released)
    - Zikula 1.0 only version, will not run with .764 or older!
    
Version 1.22  - last version for .764 or older versions of PostNuke
    - internal version

Version 1.18
    - PostNuke .8 fixes
    - fixed [#3411] XHTML Validation errors

Version 1.17
    - no new features, just internal cleanups:
    - some .762 changes
    - javscript changes
    - layout changes
    - inline styles moved to style.css

Version 1.16
    - minimum requirement now PN .760+
    - removed displayhook due to compatibility problems
      **********************************************
      **                IMPORTANT!!!              **
      **   You need to go the hooks settings      **
      **   for all modules where this transform   **
      **   hook is activated and submit this      **
      **   page in order to activte the           **
      **   display hooks too!!!                   **
      **********************************************

Version 1.15:
    - showing smilies now as display hook.

Version 1.14 AutoSmilie: 2005-04-29
	- added a Module-Var, which indicates a directory
	  All Files out of this directory (gif,jpg,pjep,png) are included automatically as
	  Smilies in this hook
	  Credits for the basic idea and work go to Bolbit (Thomas Pawlitzki, mail@2mas.de)
	  This closes FR[ 1058440 ] add Smiles
	  implemented FR [ 1135445 ] Generalise the hooks javascript
	  smilies will be added at the cursor position now
	  IMPORTANT: the textarea needs an id now which has to be passed to the pn_bbsmile module:
	  <!--[ pnmodfunc modname=pn_bbsmile type=user func=bbsmiles textfieldid='mytextfieldid' ]-->
	  shows the smilie selection

Version 1.13:
    - moved all pn_bbsmile stuff from pnForum to the hook

Version 1.12: December 19th 2004
    - code cleanups

Version 1.11: October 25th 2004
	- pnRendered admin panel

Version 1.10: July 2nd 2003
	- added alt-tags for better HTML compliance

Version 1.00: January 17th 2003
	- minor fix in smiles code (better compatibility to forum posts)

Version 0.99: January 11th 2003
	- initial release


bbsmile/html/modules/BBSmile/pninit.php
<?php
// $Id$
// ----------------------------------------------------------------------
// LICENSE
//
// This program is free software; you can redistribute it and/or
// modify it under the terms of the GNU General Public License (GPL)
// as published by the Free Software Foundation; either version 2
// of the License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// To read the license please visit http://www.gnu.org/copyleft/gpl.html
// ----------------------------------------------------------------------
// Original Author of file: Hinrich Donner
// changed to bbsmile: larsneo
// ----------------------------------------------------------------------

/**
 * @package Zikula_Utility_Modules
 * @subpackage bbsmile
 * @license http://www.gnu.org/copyleft/gpl.html
*/

/**
 * init module
*/
function bbsmile_init() {

    // Set up module variables
    //
    // - where are the smilies stored
    pnModSetVar('bbsmile', 'smiliepath',      'modules/bbsmile/pnimages/smilies');
    pnModSetVar('bbsmile', 'activate_auto',   '0');
    pnModSetVar('bbsmile', 'remove_inactive', '1');
    pnModSetVar('bbsmile', 'smiliepath_auto', 'modules/bbsmile/pnimages/smilies_auto');

    // Generate the smile array
    pnModAPIFunc('bbsmile','admin','updatesmilies',array(), true);


    // Set up module hooks
    // transform hook
    if (!pnModRegisterHook('item',
                           'transform',
                           'API',
                           'bbsmile',
                           'user',
                           'transform')) {
        return LogUtil::registerError(_BBSMILE_COULDNOTREGISTER . ' (transform hook)');
    }
    // Initialisation successful
    return true;
}

/**
 * upgrade module
*/
function bbsmile_upgrade($oldversion)
{
	switch($oldversion) {
	    case '1.13':
            pnModSetVar('pn_bbsmile', 'smiliepath',       'modules/bbsmile/pnimages/smilies');
	        pnModSetVar('pn_bbsmile', 'activate_auto',    '0');
            pnModSetVar('pn_bbsmile', 'remove_inactive',  '1');
	        pnModSetVar('pn_bbsmile', 'smiliepath_auto',  'modules/bbsmile/pnimages/smilies_auto');
	        pnModAPIFunc('pn_bbsmile','admin','updatesmilies',array());
        case '1.14':
            // display hook
            if (!pnModRegisterHook('item',
                                   'display',
                                   'GUI',
                                   'pn_bbsmile',
                                   'user',
                                   'smilies')) {
                return LogUtil::registerError(_BBSMILE_COULDNOTREGISTER . ' (display hook)');
            }
            pnModSetVar('pn_bbsmile', 'displayhook', '1');
        case '1.15':
            if (!pnModUnregisterHook('item',
                                     'display',
                                     'GUI',
                                     'pn_bbsmile',
                                     'user',
                                     'smilies')) {
                LogUtil::registerError(_BBSMILE_COULDNOTUNREGISTER . ' (display hook)');
                return '1.15';
            }
            pnModDelVar('pn_bbsmile', 'displayhook');
        case '1.17':
        case '1.18':
            // .8 only version
        case '2.0':

            pnModSetVar('bbsmile', 'smiliepath', str_replace('pn_bbsmile', 'bbsmile', pnModGetVar('pn_bbsmile', 'smiliepath')));
            pnModSetVar('bbsmile', 'smiliepath_auto', str_replace('pn_bbsmile', 'bbsmile', pnModGetVar('pn_bbsmile', 'smiliepath_auto')));
            pnModSetVar('bbsmile', 'activate_auto', pnModGetVar('pn_bbsmile', 'activate_auto'));
            pnModSetVar('bbsmile', 'remove_inactive', pnModGetVar('pn_bbsmile', 'remove_inactive'));
            $smilie_array = pnModGetVar('pn_bbsmile', 'smilie_array');
            if(@unserialize($smilie_array)!=='') {
                $smilie_array = unserialize($smilie_array);
            }
            pnModSetVar('bbsmile', 'smilie_array', $smilie_array);

            pnModDelVar('pn_bbsmile');

    		// update hooks
    		$pntables = pnDBGetTables();
    		$hookstable  = $pntables['hooks'];
    		$hookscolumn = $pntables['hooks_column'];
    		$sql = 'UPDATE ' . $hookstable . ' SET ' . $hookscolumn['smodule'] . '=\'bbsmile\' WHERE ' . $hookscolumn['smodule'] . '=\'pn_bbsmile\'';
    		$res = DBUtil::executeSQL ($sql);
    		if ($res === false) {
        		LogUtil::registerError(_BBSMILE_FAILEDTOUPGRADEHOOK . ' (smodule)');
        		return '2.0';
    		}

    		$sql = 'UPDATE ' . $hookstable . ' SET ' . $hookscolumn['tmodule'] . '=\'bbsmile\' WHERE ' . $hookscolumn['tmodule'] . '=\'pn_bbsmile\'';
    		$res   = DBUtil::executeSQL ($sql);
    		if ($res === false) {
        		LogUtil::registerError(_BBSMILE_FAILEDTOUPGRADEHOOK . ' (tmodule)');
        		return '2.0';
    		}

        default:
            break;
    }
    return true;
}

/**
 * delete module
*/
function bbsmile_delete() {

    // Remove module hooks
    if (!pnModUnregisterHook('item',
                             'transform',
                             'API',
                             'bbsmile',
                             'user',
                             'transform')) {
        return LogUtil::registerError(_BBSMILE_COULDNOTUNREGISTER . ' (transform hook)');
    }

    // Remove module variables
    pnModDelVar('bbsmile');

    // Deletion successful
    return true;
}
